### PR TITLE
Add a check on frontend dependencies

### PR DESCRIPTION
**Pull request by Bernardstanislas**
This let us enforce some standards, such as the components standard: a React component should live in the `components` folder only if it is used more than once in the app.
We'll be able to add more rules in the futures, such as the fractal design rule.

### CODE REVIEW

File: .dependency-cruiser.js [hunk 1]
```diff
@@ -0,0 +1,294 @@
+/** @type {import('dependency-cruiser').IConfiguration} */
+module.exports = {
+  forbidden: [
+    {
+      name: "no-orphans",
+      comment:
```
Comment (AnthonyBuisset) [inline .dependency-cruiser.js:6]: sometimes we create orphans components during development, like [here](https://linear.app/onlydust/issue/B-763/create-new-githubissue-component).
It is temporary as aimed to be used in a later task.

How should we handle such cases ?

**Review by AnthonyBuisset — APPROVED** [review]
Nice initiative !

File: .dependency-cruiser.js (hunk 1 above)
Comment (Bernardstanislas) [reply .dependency-cruiser.js:6]: I think I'm in favor of explicitly whitelisting such component when it is temporary an orphan, that would leave the elephant in the room and let us remove it easily afterwards

File: .github/workflows/frontend-ci.yml [hunk 2]
```diff
@@ -41,6 +41,17 @@ jobs:
       - run: yarn install --frozen-lockfile
       - run: yarn lint
 
+  dependencies:
```
[the remark this answers is not in the document]
Comment (Bernardstanislas) [reply .github/workflows/frontend-ci.yml:44]: I don't know, I think it is a question of philosophy of the epic branch. From what I see, we are facing the problem of leaving some tasks for the end of the epic development, whereas if we enforce PR are as clean on an epic branch as they would be on `main`, that would help reducing this "end of epic garbage collection"

File: frontend/src/pages/ProjectDetails/Contributors/ContributorsTable/Headers.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { useIntl } from "src/hooks/useIntl";
-import HeaderLine from "../Table/HeaderLine";
-import HeaderCell, { HeaderCellWidth } from "../Table/HeaderCell";
+import HeaderLine from "../../../../components/Table/HeaderLine";
```
[the remark this answers is not in the document]
Comment (Bernardstanislas) [reply frontend/src/pages/ProjectDetails/Contributors/ContributorsTable/Headers.tsx:2]: agreed, I'm changing this

File: frontend/src/pages/ProjectDetails/Sidebar/View.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { Listbox } from "@headlessui/react";
 import UpDownChevrons from "src/assets/icons/UpDownChevrons";
 import { RoutePaths } from "src/App";
-import BackLink from "src/components/BackLink";
+import BackLink from "src/pages/ProjectDetails/Sidebar/BackLink";
```
[the remark this answers is not in the document]
Comment (Bernardstanislas) [reply frontend/src/pages/ProjectDetails/Sidebar/View.tsx:4]: Yes!

File: .dependency-cruiser.js (hunk 1 above)
Comment (AnthonyBuisset) [reply .dependency-cruiser.js:6]: ok let's start like this and adapt later if needed

File: .github/workflows/frontend-ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (AnthonyBuisset) [reply .github/workflows/frontend-ci.yml:44]: ok !

**Pull request merged**